tests/test_weaver_snapshot_serving.py

```
@@ -0,0 +1,389 @@
+    # the test would pass on a port that is simply free.
+    with (
+        socket.socket(socket.AF_INET, socket.SOCK_STREAM) as bare,
+        pytest.raises(OSError, match="in use"),
+    ):
+        bare.bind(("127.0.0.1", port))
+
+    serving._refuse_occupied_port(port)
+
+
+def test_a_kernel_assigned_port_leaves_no_lock_file_behind(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
```

Ruff (0.16.2)

[warning] Unused function argument: `tmp_path`

(ARG001)